src/Apps/W1/Subscription Billing/Test/UBB/UsageBasedBillingTest.Codeunit.al

```diff
@@ -1730,6 +1730,179 @@ codeunit 148153 "Usage Based Billing Test"
             Assert.AreEqual(BillingLine."Service Object Quantity", UsageDataBilling.Quantity, 'Billing Line Quantity should be equal to Usage Data Billing Quantity');
         until BillingLine.Next() = 0;
     end;
+
+    [HandlerFunctions('ExchangeRateSelectionModalPageHandler,CreateCustomerBillingDocumentPageHandler,MessageHandler')]
+    [Test]
+    procedure DeleteUsageDataBillingLineWhenRelatedSalesCrMemoLineIsDeleted()
+    begin
+        // [SCENARIO] Creating a corrective credit memo for a contract with two usage-based service commitments, deleting one line, and posting the memo should only create a new usage data billing line for the credited line.
+        // [GIVEN] A customer contract with two usage-based service commitments, both invoiced
+        ResetAll();
+        CreateUsageDataBilling("Usage Based Pricing"::"Fixed Quantity", LibraryRandom.RandDec(10, 2));
+        PostDocument := true;
+        UsageDataImport.ProcessUsageDataImport(UsageDataImport, Enum::"Processing Step"::"Process Usage Data Billing");
+        UsageDataImport.TestField("Processing Status", "Processing Status"::Ok);
+        UsageDataImport.CollectCustomerContractsAndCreateInvoices(UsageDataImport);
+        FilterUsageDataBillingOnUsageDataImport(UsageDataImport."Entry No.", "Service Partner"::Customer, UsageDataBilling."Document Type"::"Posted Invoice");
+        UsageDataBilling.FindSet();
+
+        SalesInvoiceHeader.Get(UsageDataBilling."Document No.");
+        CorrectPostedSalesInvoice.CreateCreditMemoCopyDocument(SalesInvoiceHeader, SalesCrMemoHeader);
+
+        SalesLine.Reset();
+        SalesLine.SetRange("Document Type", "Sales Document Type"::"Credit Memo");
+        SalesLine.SetRange("Document No.", SalesCrMemoHeader."No.");
+        SalesLine.SetRange(Type, "Sales Line Type"::Item);
+        SalesLine.FindFirst();
+
+        // Delete the first line (simulate user action)
+        SalesLine.Delete(true);
+
+        // [THEN] Only the credited line should have a new usage data billing line 
+        // Check usage data billing lines for the contract
+        UsageDataBilling.Reset();
+        UsageDataBilling.SetRange("Document Type", UsageDataBilling."Document Type"::"Credit Memo");
+        UsageDataBilling.SetRange("Document No.", SalesCrMemoHeader."No.");
+        UsageDataBilling.SetRange("Document Line No.", SalesLine."Line No.");
+        Assert.RecordIsEmpty(UsageDataBilling);
+    end;
+
+    [HandlerFunctions('ExchangeRateSelectionModalPageHandler,CreateCustomerBillingDocumentPageHandler,MessageHandler')]
+    [Test]
+    procedure DeleteUsageDataBillingLineWhenRelatedSalesInvoiceLineIsDeleted()
```

Comment:
The function name is misleading as the Scenario states that some fields in Usage Data Billing are cleared but the record is not deleted. I guess also that the Test should not check if there are not Usage Data Billings for the Invoice Line but if the Usage Data Billing has been cleared.